addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +54,208 @@ def _get_move_line_count(self):
-            if next_statements_to_recompute:
-                self.env.add_to_compute(self._fields['previous_statement_id'], next_statements_to_recompute)
-                next_statements_to_recompute.modified(['previous_statement_id'])
-        return res
-
-    @api.model_create_multi
-    def create(self, values):
-        res = super(AccountBankStatement, self).create(values)
-        # Upon bank stmt creation, it is possible that the statement is inserted between two other statements and not at the end
-        # In that case, we have to search for statement that are pointing to the same previous_statement_id as ourselve in order to
-        # change their previous_statement_id to us. This is done by marking the field 'previous_statement_id' to be recomputed for such records.
-        # Note that marking the field is not enough as we also have to recompute all its other fields that are depending on 'previous_statement_id'
-        # hence the need to call modified afterwards.
-        # The reason we are doing this here and not in a compute field is that it is not easy to write dependencies for such field.
-        next_statements_to_recompute = self.search([('previous_statement_id', 'in', [st.previous_statement_id.id for st in res]), ('id', 'not in', res.ids), ('journal_id', 'in', res.journal_id.ids)])
-        if next_statements_to_recompute:
-            self.env.add_to_compute(self._fields['previous_statement_id'], next_statements_to_recompute)
-            next_statements_to_recompute.modified(['previous_statement_id'])
-        return res
+    attachment_ids = fields.Many2many(
+        comodel_name='ir.attachment',
+        compute='_compute_attachment_ids', inverse='_inverse_attachment_ids'
+    )
 
     @api.depends('line_ids.is_reconciled')
     def _compute_all_lines_reconciled(self):
         for statement in self:
             statement.all_lines_reconciled = all(st_line.is_reconciled for st_line in statement.line_ids)
 
-    @api.onchange('journal_id')
-    def _onchange_journal_id(self):
-        for st_line in self.line_ids:
-            st_line.journal_id = self.journal_id
-            st_line.currency_id = self.journal_id.currency_id or self.company_id.currency_id
-
-    def _check_balance_end_real_same_as_computed(self):
-        """ Check the balance_end_real (encoded manually by the user) is equals to the balance_end (computed by odoo). """
-        return self._check_cash_balance_end_real_same_as_computed() and self._check_bank_balance_end_real_same_as_computed()
-
-    def _check_cash_balance_end_real_same_as_computed(self):
-        """ Check the balance_end_real (encoded manually by the user) is equals to the balance_end (computed by odoo).
-            For a cash statement, if there is a difference, the different is set automatically to a profit/loss account.
+    @api.model
+    def _garbage_clean_statements(self):
```

Comment:
garbage "collect" is a more common term

Reply:
Why is it called in a cron? Isn't it easy to check when unlinking the lines that you need to unlink the related statement lines?

Reply:
Think the user has removed the statement from e.g. line 21 because he wanted to use it in line 22, if we erase it immediately, after unsetting it on line 21, the statement is erased.

Reply:
There was no direct way to delete an [empty] statement before, so we had to clean it automatically, but currently with the latest spec, we keep the list view of the statements, so they can be deleted from there. 
I think now we don't really have to automate it.